app/utils/ErrorHandler.scala

```diff
@@ -37,10 +37,13 @@ trait ErrorHandler {
       Future.failed(new BadRequestException(e.message))
     case e: NotFoundException =>
       Future.failed(new NotFoundException(e.message))
-    case e: Upstream4xxResponse =>
-      Future.failed(Upstream4xxResponse(e.message, e.upstreamResponseCode, e.reportAs))
-    case e: Upstream5xxResponse =>
-      Future.failed(Upstream5xxResponse(e.message, e.upstreamResponseCode, e.reportAs))
+    case e: UpstreamErrorResponse =>
```

Comment:
No need for case match because both are doing the same thing.. Can be combined into a single case

Reply:
Done